### PR TITLE
Add `onAfterDestroy` callback description to react docs

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Add `onAfterDestroy` callback description to React integration docs.

### Why has been this callback introduced?

1. Editor uses async destroy method that does not work super fine with React lifecycle. 
2. Editor is most of the time destroyed _after_ unmounting of component or changing its ID property. 
3. This callback makes sure that all side effects (such as added editable HTML DOM nodes) are removed _after_ destroying all plugins attached to the editor. 
4. `onAfterDestroy` is being batched, so in other words rapid `id` change (10 in row) will trigger only ~1 call of this callback which prevents race conditions in `Strict Mode`. 

### Additional information

1. The `onAfterDestroy` callback has been added in this [commit](https://github.com/ckeditor/ckeditor5-react/commit/ce6cad06e7a847b48189ddfaeed57921ce314874#diff-17a78b4174b47153db11ac4738dfbb3608e1b36b02c7d70d86589722797e8515R181) in our React integration.
5. It's not released yet at this moment (20.05.2024).

### CODE REVIEW

**Comment by Mati365** [comment]
Thank you @godai78. 

@pomek Can you merge it after release of React integration? 

fyi @scofalik

File: docs/installation/integrations/react.md
Comment (pomek) [inline docs/installation/integrations/react.md:234]:
```suggestion
					onAfterDestroy={ editor => {
						editor.ui.view.toolbar.element.remove();
					} }
```

File: docs/installation/integrations/react.md
Comment (pomek) [inline docs/installation/integrations/react.md:165]:
```suggestion
* `onAfterDestroy` &ndash; A function called after the successful destruction of an editor instance rendered by the component. This callback is also triggered after the editor has been reinitialized after an error. The component is not guaranteed to be mounted when this function is called.
```

Minor updates:

* `s/invoked/called` - otherwise, we should update other callbacks too. 
* Simplifications.

**Review by pomek — COMMENTED** [review]
Except for the mentioned things, please target the PR to the `#stable` branch if we aim to deploy these changes without waiting until the next release.

**Comment by Mati365** [comment]
@pomek It can be deployed in normal flow I think. There is no need to deploy it instantly after merge because this callback is not critical to work with integration (especially because it was not present for years). Thanks for suggestion to docs anyway.

File: docs/installation/integrations/react.md
Comment (Mati365) [reply docs/installation/integrations/react.md:234]: I'm not sure if we should include this `onAfterDestroy` example with `editor.ui.view.toolbar.element.remove();`. It really depends on the way the users mounted `toolbar`. If it's somewhere outside of React box then it's fine but if it's in div that is managed by React then it will lead to crashes, probably. I'll leave there a comment that only external things should be removed here.

Comment (scofalik) [reply docs/installation/integrations/react.md:234]: For more context: this code sample is for Decoupled Editor. This editor is characteristic in the way that it is the integrator's responsibility to insert the toolbar where they want to.

**Comment by scofalik** [comment]
LGTM, you can merge it. As far as I understand, neither the new callback nor the docs for the callback are going to be released in this release (which should happen very shortly)?

**Review by scofalik — APPROVED** [review]

**Review by pomek — APPROVED** [review]

**Comment by pomek** [comment]
The required package has been released (https://github.com/ckeditor/ckeditor5-react/releases/tag/v7.0.0).

**Comment by godai78** [comment]
We have ported these changes to new installation docs, which we plan on merging to `master` sometime soon, so this will be wvisible on nightly.

**Pull request merged**
